lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -357,8 +357,16 @@ SILInstruction *TempRValueOptPass::getLastUseWhileSourceIsNotModified(
     if (numLoadsFound == useInsts.size()) {
       // Function calls are an exception: in a called function a potential
       // modification of copySrc could occur _before_ the read of the temporary.
+      auto isLetObject = [](SILValue copySrc) {
+        SILValue object = getUnderlyingObject(copySrc);
+        if (auto *ASI = dyn_cast<AllocStackInst>(object)) {
+          if (ASI->isLet())
+            return true;
+        }
+        return false;
+      };
       if ((FullApplySite::isa(inst) || isa<YieldInst>(inst)) &&
-          aa->mayWriteToMemory(inst, copySrc)) {
+          !isLetObject(copySrc) && aa->mayWriteToMemory(inst, copySrc)) {
```

Comment:
Now, you can just use `bool swift::isLetAddress(SILValue address)`. That uses the canonical AccessBase utility. It's a bit less ad-hoc and you don't need to DIY.

Reply:
@meg-gupta pointed out that if you just check `AllocStack::isLet()` inside `isLetAddress`, then you shouldn't need to touch this pass at all.